tracker/tracker_test.go

```diff
@@ -767,9 +767,12 @@ func TestTooMuchDataRequested(t *testing.T) {
 	tt, _ := NewTracker(mm,
 		WithFilter(&FilterConfig{Async: true}),
 	)
-	if err := tt.Sync(context.Background()); err != nil {
-		t.Fatal(err)
-	}
+	go func() {
+		if err := tt.Sync(context.Background()); err != nil {
+			t.Fatal(err)
+		}
+	}()
+	time.Sleep(3 * time.Second)
```

Comment:
Any advice on appropriate wait here? The previous impl had synchronous batchSync() which we could pull out of Sync() to emulate previous test behaviour

Reply:
I think for this example `BatchSync` function might be enough to make it work.